### PR TITLE
feat: enable output to be JSON encoded

### PR DESCRIPTION
**Pull request by js-timbirkett**
TL;DR - Adds a cli clag `--json` to output the changelog as a JSON encoded string.

## What does this do / why do we need it?

`git-chglog` is brilliant and outputs wonderfully formatted changelogs to STDOUT or a file... We'd like to start piping changelogs into MS Teams, Slack and other web APIs on release/deployment that only accept valid formatted JSON.

We _could_ use `sed`, `awk` etc... or write a python or ruby script to pipe output through but it might nice to be able to get a JSON sting back straight from `git-chglog`

## How this PR fixes the problem?

Allows the user to specify a cli flag: `--json` to get encoded JSON. 

## What should your reviewer look out for in this PR?

This is an initial commit, I still need to add:
- [ ] Tests
- [ ] Docs

and it's an early commit for discussion or closure if this feature isn't desirable.  

## Check lists

* [ ] Test passed
* [ ] Coding style (indentation, etc)

### CODE REVIEW

**Comment by js-timbirkett** [comment]
Will look at adding tests to 📈 the tests soon.

**Comment by mavogel** [comment]
please rebase @js-timbirkett

**Review by neeraj00890 — APPROVED** [review]
